arch/xtensa/src/esp32/Kconfig

```diff
@@ -362,11 +362,53 @@ endmenu # I2C configuration
 menu "SPI configuration"
 	depends on ESP32_SPI
 
+config ESP32_SPI_SWCS
+	bool "SPI software CS"
+	default y
+	---help---
+		Use SPI software CS.
+
+config ESP32_SPI_UDCS
+	bool "User defined CS"
+	default n
+	depends on ESP32_SPI_SWCS
+	---help---
+		Use user defined CS.
+
+choice
+	prompt "DMA owner"
+	default ESP32_SPI2_DMA
+	depends on ESP32_SPI2 || ESP32_SPI3
+	---help---
+		Select which SPI owns DMA, only one SPI DMA is available, so
+		only one SPI can use DMA.
```

Comment:
By this you mean DMA controller or DMA channels?
From the manual:
>two DMA channels are shared by SPI1, SPI2 and SPI3 controllers. Each DMA channel can be used by
any one SPI controller at any given time.

It looks like the code expects to have only one channel (channel 2) available.  Am I looking at an old document?

Reply:
Yes, you are right. I mean DMA channel here, not just DMA.

ESP32 has 4 SPI hardware controllers, SPI0 for code flash, SPI1 is a little different from SPI2 and SPI3, because it has an arbiter connecting to IO, add its I/O pins are same as SPI0, so I have not added SPI1 driver.

My original idea was keeping a channel for SPI1, and driver only uses one channel. So it is what it is now.

Reply:
But maybe letting SPI2 and SPI3 use both these DMA channel is better design.

Reply:
Ah ok, you wanted to dedicate one DMA channel for SPI1.  I think we can still manage that through Kconfig and have the two channels available for SPI2 and SPI3 in case SPI1 is not used, but I understand if this is how things are set now and can be improved for another iteration.